### PR TITLE
Use JSON.parse when safe parse option isn't explicilty enabled

### PR DESCRIPTION
**Pull request by tomasz-szymanek**
https://github.com/elastic/elasticsearch-js/issues/2303

### CODE REVIEW

**Comment by tomasz-szymanek** [comment]
[Please_DocuSign_Individual_Contributor_Licen.pdf](https://github.com/user-attachments/files/17028285/Please_DocuSign_Individual_Contributor_Licen.pdf)

I have signed it.

**Comment by JoshMock** [comment]
When double-checking [the docs](https://github.com/fastify/secure-json-parse?tab=readme-ov-file#sjsonparsetext-reviver-options) for secure-json-parse, it states clearly that it uses `JSON.parse` behind the scenes when `ignore` is used for `protoAction` or `constructorAction`. Their [published benchmarks](https://github.com/fastify/secure-json-parse?tab=readme-ov-file#benchmarks) make it pretty clear that there is no notable performance drop when `ignore` is set to true for both. I cloned the repo and ran the benchmarks myself to verify this.

Basically, this PR is redundant, and the change you made to the `SerializerOptions` type is, arguably, a breaking one.
